base/dates/io.jl

```diff
@@ -252,13 +253,21 @@ const SLOT_RULE = Dict{Char, Type}(
     's' => Millisecond,
 )
 
-slot_order(::Type{Date}) = (Year, Month, Day)
-slot_order(::Type{DateTime}) = (Year, Month, Day, Hour, Minute, Second, Millisecond)
-
-slot_defaults(::Type{Date}) = map(Int64, (1, 1, 1))
-slot_defaults(::Type{DateTime}) = map(Int64, (1, 1, 1, 0, 0, 0, 0))
+const FORMAT_DEFAULTS = Dict{Type, Any}(
+    Year => Int64(1),
+    Month => Int64(1),
+    DayOfWeekToken => Int64(0),
+    Day => Int64(1),
+    Hour => Int64(0),
+    Minute => Int64(0),
+    Second => Int64(0),
+    Millisecond => Int64(0),
+)
 
-slot_types{T<:TimeType}(::Type{T}) = typeof(slot_defaults(T))
+const FORMAT_TRANSLATIONS = Dict{Type{<:TimeType}, Tuple}(
+    Date => (Year, Month, Day),
+    DateTime => (Year, Month, Day, Hour, Minute, Second, Millisecond),
+)
```

Comment:
Originally these were defined through methods (which I prefer):
```julia
slot_order(::Type{DateTime}) = (Year, Month, Day, Hour, Minute, Second, Millisecond)
```
Unfortunately this cannot be used here since generated functions cannot call methods defined after the generated function definition.